### PR TITLE
Api redux

### PR DESCRIPTION
**Pull request by ANTOINE1128**
## In this PR I have done the following :
 - Make use of the Bookstore API as the backend.
- Used Axios with createAsyncThunk to make the HTTP requests to the backend
- Fetch  books from the Bookstore API after the component has rendered
- Refactored add book and remove book features to persist the changes in the server
- The application behaves the same way after the refactoring

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
 import React from 'react';
-import { Routes, Route } from 'react-router-dom';
-import Navbar from './components/Navbar';
-import Categories from './components/Categories';
-import RenderBooks from './components/RenderBooks';
+import './App.css';
+import { Route, Routes } from 'react-router-dom';
+import Navigation from './components/navigation';
+import BooksDisplay from './components/booksDisplay';
+import Categories from './components/categories';
```
Comment (FrederickMih) [inline src/App.js]: - [ ] Based on the requirements of this project, the addBook and removeBook features should still function as before and the app should open in the browser. I got the console error on the screenshot below. Kindly make sure to reference the correct file(Categories not categories)
Screenshot from 2023-06-23 19-17-05

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @ANTOINE1128, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights :green_circle: 

- Great job for preventing side effects in your component by using useEffect hook :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @ANTOINE1128 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failed Eslint checks. remember it helps identify problematic patterns found in your code. please kindly fix that.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @ANTOINE1128,

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ You used createAsyncThiunk and Axios correctly 👍🏽
✅ Add and remove book features work fine as expected 👍🏽
✅ The app runs fine 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**